apptest.go

```diff
@@ -497,15 +495,31 @@ func (a *AppSetup) createUserValuesConfigMap(ctx context.Context, name, namespac
 		},
 	}
 
-	_, err := a.k8sClient.CoreV1().ConfigMaps(namespace).Create(ctx, configMap, metav1.CreateOptions{})
-	if apierrors.IsAlreadyExists(err) {
-		a.logger.Debugf(ctx, "already created configmap '%s/%s'", namespace, name)
-	} else if err != nil {
+	_, err := a.k8sClient.CoreV1().ConfigMaps(namespace).Get(ctx, name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		a.logger.Debugf(ctx, "creating configmap '%s/%s'", namespace, name)
+
+		_, err := a.k8sClient.CoreV1().ConfigMaps(namespace).Create(ctx, configMap, metav1.CreateOptions{})
+		if apierrors.IsAlreadyExists(err) {
+			a.logger.Debugf(ctx, "already created configmap '%s/%s'", namespace, name)
```

Comment:
I think we need to update when configmap is already exists, doesn't it?

Reply:
Sorry about the late reply. We get the CM first and then match on apierrors.IsNotFound(err) to decide whether we should create or update. 

So I think this is already covered. WDYT?